tanner/emulators/sqli.py

```diff
@@ -70,9 +70,17 @@ def map_query(self, query):
         db_query = None
         param = query[0][0]
         param_value = query[0][1].replace('\'', ' ')
-        tables = [k for k, v in self.query_map.items() if query[0][0] in v]
+        tables = []
+        for table, columns in self.query_map.items():
+            for column in columns: 
+                if query[0][0] == column['name']:
+                    tables.append(dict(table_name=table, column=column))
+
         if tables:
-            db_query = 'SELECT * from ' + tables[0] + ' WHERE ' + param + '=' + param_value + ';'
+            if tables[0]['column']['type'] == 'INTEGER':
+                db_query = 'SELECT * from ' + tables[0]['table_name'] + ' WHERE ' + param + '=' + param_value + ';'
```

Comment:
If concatenated, q = `'SELECT * FROM users where name="'+some_name_var+ '";'` and an attack would make it `SELECT * FROM users where name=random" or 1=1 --";`

If parameterized using python SQLite, q=`'SELECT * FROM users where name=?'`
Now if we try to inject `random" or 1=1 --`, it won't produce anything. I'm NOT sure about how python SQLite interpret it but I guess it bypass quotes something as done by `mysql_real_escape_string` in php

Though we can use something like q = `'SELECT * FROM users where name="%s"'`